eng/ci/templates/jobs/build-and-test.yml

```diff
@@ -0,0 +1,101 @@
+parameters:
+  - name: SkipBuildArtifacts
+    type: boolean
+    default: false
+
+jobs:
+
+- job: BuildAndTest
+  displayName: Build and Test
+
+  pool:
+    name: 1es-pool-azfunc
+    image: 1es-windows-2022
+    os: windows
+
+  templateContext:
+    outputParentDirectory: $(Build.ArtifactStagingDirectory)
+    outputs:
+    - ${{ if eq(parameters.SkipBuildArtifacts, false) }}:
+      - output: pipelineArtifact
+        displayName: Publish artifacts
+        path: $(Build.ArtifactStagingDirectory)/pkg
+        artifact: drop
+
+  steps:
+  - task: UseDotNet@2
+    displayName: Install .NET 6
+    inputs:
+      version: 6.x
+
+  - task: DotNetCoreCLI@2
+    displayName: 'Build'
+    inputs:
+      command: 'build'
+      arguments: -c Release
+      projects: |
+        src\**\Microsoft.NET.Sdk.Functions.MSBuild.csproj
+        src\**\Microsoft.NET.Sdk.Functions.Generator.csproj
+        src\**\Microsoft.NET.Sdk.Functions.csproj
+
+  - task: DotNetCoreCLI@2
+    displayName: Unit Tests
+    inputs:
+      command: test
```

Comment:
this will be testing in debug. Add arguments for `--no-build -c Release`

Reply:
The tests are not running if I am adding these arguments. Here is the sample run for reference- [link](https://dev.azure.com/azfunc/internal/_build/results?buildId=173607&view=results). Only 6 tests ran here while there are total 71 tests. Original pipeline also does not have these arguments for unit tests - [link](https://github.com/Azure/azure-functions-vs-build-sdk/blob/e9cbe68782d021e0c2c5a51ae132a90ccaacbb5b/azure-pipelines.yml#L35)

Reply:
Interesting. I'll approve and unblock this for now, but lets follow up after to see what is going on